Find a functionally equivalent code:android.text.Layout.measurePara:COMMENT
Method Modifier: default     static      
Comment:/* package */

@@ -16,21 +16,30 @@
         int len = mt.mLen;
         boolean hasTabs = false;
         TabStops tabStops = null;
+        // leading margins should be taken into account when measuring a paragraph
+        int margin = 0;
+        if (text instanceof Spanned) {
+            Spanned spanned = (Spanned) text;
+            LeadingMarginSpan[] spans = getParagraphSpans(spanned, start, end, LeadingMarginSpan.class);
+            for (LeadingMarginSpan lms : spans) {
+                margin += lms.getLeadingMargin(true);
+            }
+        }
         for (int i = 0; i < len; ++i) {
             if (chars[i] == '\t') {
                 hasTabs = true;
@@ -31,37 +40,46 @@
             }
         }
         tl.set(paint, text, start, end, dir, directions, hasTabs, tabStops);
-        return tl.metrics(null);
+        return margin + tl.metrics(null);
     } finally {
         TextLine.recycle(tl);
         MeasuredText.recycle(mt);

